File: scala/api/src/main/scala/org/taymyr/lagom/scaladsl/openapi/LagomError.scala
package org.taymyr.lagom.scaladsl.openapi

import io.swagger.v3.oas.annotations.media.Schema

import scala.annotation.meta.field
import scala.beans.BeanProperty

@Schema(requiredProperties = Array("name", "detail"))
case class LagomError(
    @(Schema @field)(description = "Exception name")
    @BeanProperty name: String,
    @(Schema @field)(description = "Exception detail message")
    @BeanProperty detail: String
)

File: java/api/src/main/kotlin/org/taymyr/lagom/javadsl/openapi/LagomError.kt
package org.taymyr.lagom.javadsl.openapi

import io.swagger.v3.oas.annotations.media.Schema

@Schema(requiredProperties = ["name", "detail"])
class LagomError {
    @Schema(description = "Exception name")
    fun getName() = ""
    @Schema(description = "Exception detail message")
    fun getDetail() = ""
}
